titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/KubeNotificationProcessor.java

```diff
@@ -290,8 +290,13 @@ static Task updateTaskStatus(PodWrapper podWrapper,
                 .build();
 
         Task fixedTask = fillInMissingStates(podWrapper, updatedTask);
-        Task taskWithPlacementData = JobManagerUtil.attachPlacementData(fixedTask, executorDetailsOpt);
-        Task taskWithNodeMetadata = node.map(n -> attachNodeMetadata(taskWithPlacementData, n)).orElse(taskWithPlacementData);
+        Task taskWithExecutorData;
+        if (executorDetailsOpt == null) {
```

Comment:
`executorDetailsOpt == null || executorDetailsOpt.isEmpty()`

Reply:
There is no method like that on this object (because it is optional?)

There is a `isPresent`, but it just checks for nullness too.